grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/state/AtomicState.java

```diff
@@ -41,85 +40,100 @@
  * @author Kasper Piskorski
  *
  */
-public class AtomicState extends QueryState{
-
-    private final ReasonerAtomicQuery query;
-    private final Iterator<Answer> dbIterator;
-    private final Iterator<Pair<InferenceRule, Unifier>> ruleIterator;
-
-    private InferenceRule currentRule = null;
-    private final MultiUnifier cacheUnifier;
+@SuppressFBWarnings("BC_UNCONFIRMED_CAST_OF_RETURN_VALUE")
```

Comment:
Why did you have to suppress this? What's causing it?

Reply:
call to getQuery()